docs/arch/adr-15.md

```diff
@@ -0,0 +1,40 @@
+# ADR 15: Drawing tools API
+
+June 19, 2019
+
+## Context
+
+The way the drawing tools currently function on Panoptes-Front-End (PFE) have numerous issues including:
+
+- Re-rendering the DOM more than necessary
+- Updating the classification annotation on each touch or click event
+
+Both of these can cause significant performance issues.
+
+- The separation concerns are not clear between components and stores. Multiple components can update what is being rendered making it hard to debug or add new features to. 
+
+## Decision
+
+What we do not want to do:
+- Re-render every time a click or touch event happens
```

Comment:
```suggestion
- Re-render on every pointer or touch event.
```

Reply:
The current mark will have to render on each event, since we're changing it, but we want to avoid re-rendering the whole drawing surface and any static marks.

`React.memo()` and pure components, with sensibly designed props, will also help here.